workflows/nf_chai/main.nf

```diff
@@ -16,12 +16,31 @@ include { softwareVersionsToYAML } from '../../subworkflows/nf-core/utils_nfcore
 workflow NF_CHAI {
 
     take:
-    ch_fasta // channel: fasta file read in from --input
+    fasta_file       //  string: path to fasta file read provided via --input parameter
+    msa_dir          //  string: path to MSA directory read provided via --msa_dir parameter
+    constraints_file //  string: path to constraints file read provided via --constraints parameter
+    model_dir        //  string: path to model directory read provided via --model_dir parameter
 
     main:
 
     ch_versions = Channel.empty()
 
+    // Input channel for FASTA files
+    Channel
+        .fromPath(fasta_file)
```

Comment:
I already have a prototype implementation for this that reads in and chunks the FASTA file up to parallelise. Didn't add it here because i don't know if it will give us significant gains in performance. Something to ask the Chai team.